File: src/main/java/br/com/capela/amicao/modelo/cliente/Endereco.java
package br.com.capela.amicao.modelo.cliente;

import org.apache.commons.lang.builder.EqualsBuilder;
import org.apache.commons.lang.builder.HashCodeBuilder;

import br.com.capela.amicao.modelo.base.Entidade;

public class Endereco extends Entidade<Endereco> {

	private String logradouro;
	private String numero;
	private String bairro;
	private String cidade;
	private String cep;

	@Override
	public Boolean isMesmaEntidade(Endereco outro) {
		return new EqualsBuilder().append(logradouro, outro.logradouro)
				.append(numero, outro.numero).append(bairro, outro.bairro).append(cep, outro.cep)
				.isEquals();
	}

	@Override
	public int hashCode() {
		return new HashCodeBuilder().append(logradouro).append(numero).append(bairro).append(cep)
				.toHashCode();
	}
}

File: src/main/scala/br/com/capela/amicao/modelo/cliente/Endereco.scala
package br.com.capela.amicao.modelo.cliente;

import org.apache.commons.lang.builder.EqualsBuilder;
import org.apache.commons.lang.builder.HashCodeBuilder;

import br.com.capela.amicao.modelo.base.Entidade;
import br.com.capela.amicao.modelo.pet.Pet;

import java.io.Serializable;

import javax.persistence._

import scala.reflect._

@Entity
class Endereco extends Entidade[Endereco] {
	
	@Column{val name="logradouro"}
    @BeanProperty var logradouro:String = _
    
    @Column{val name="numero"}
    @BeanProperty var numero:String = _
    
    @Column{val name="bairro"}
    @BeanProperty var bairro:String = _
    
    @Column{val name="cidade"}
    @BeanProperty var cidade:String = _
    
    @Column{val name="cep"}
    @BeanProperty var cep:String = _
	
	def isMesmaEntidade(outro:Endereco):Boolean =  {
		new EqualsBuilder()
			.append(numero, outro.numero)
			.append(bairro, outro.bairro)
			.append(cep, outro.cep)
			.isEquals();
	}

	override def hashCode():Int = {
		new HashCodeBuilder()
			.append(logradouro)
			.append(numero)
			.append(bairro)
			.append(cep)
			.toHashCode();
	}
}
